### ISSUE TITLE
Is not possible to listen events on ShadowRoot

### ISSUE THREAD

**Issue by blikblum** [jquery/jquery#4317]
### Description ###

When trying to listen events on a ShadowRoot element using `on`, the handler is not fired.
The native addEventListener works

Minimal example:
```javascript
const el = $("#test")[0];
el.attachShadow({ mode: "open" });
el.shadowRoot.innerHTML = `<div class=".inner">Inner content</div>`;
$(el.shadowRoot).on("click", ".inner", () => {
  console.log("jquery - shadow inner clicked");
});
```

The issue is that the element data returned by dataPriv is always empty because of [this check](https://github.com/jquery/jquery/blob/master/src/data/var/acceptData.js#L16)

ShadowRoot elements have nodeType === 11

The [issue](https://bugs.jquery.com/ticket/8335) [pointed](https://github.com/jquery/jquery/blob/e743cbd28553267f955f71ea7248377915613fd9/src/data/Data.js#L28) as reason for the check seems not valid for ShadowRoot elements which are supported only on modern browsers

So, changing the check for 
```javascript
return owner.nodeType === 1 || owner.nodeType === 9 || owner.nodeType === 11 || !( +owner.nodeType );
```

should work

### Link to test case ###
https://codepen.io/blikblum/pen/QogVgO?editors=1111

**Comment by dmethvin** [comment]
I agree with the bug but not the solution. We shouldn't be adding event listeners to any `nodeType===11` which includes `DocumentFragment`. For example, if a jQuery event handler was attached to a `DocumentFragment` and the DF was later inserted in to the document, we wouldn't clean up internal data structures.

**Comment by blikblum** [comment]
BTW: the underlying issue is not specific for events. It also affects [setting data](https://codepen.io/blikblum/pen/zbMpaW?editors=1111).

AFAICS the memory leak that prevented attaching data to non elements does not occurs with current implementation of jQuery and >= IE8, so should not be a problem.

If attaching to `DocumentFragment` is still a concern, should be acceptable a more specific check to restrict to `ShadowRoot`?

Something like:
```javascript
return owner.nodeType === 1 || owner.nodeType === 9 || (owner.nodeType === 11 && owner.host) || !( +owner.nodeType );
```

**Comment by dmethvin** [comment]
Yeah that might work!

**Comment by extempl** [comment]
Bump?

**Comment by dmethvin** [comment]
Hi @extempl, a proposed solution is above. Would you like to submit a pull request?

**Comment by Kevinjohn** [comment]
I'd like to help move this forward, since it's still labeled **help wanted**. 

I don't know if @dmethvin is still active / owning this, and wanted to check first.
I'd be glad to put together a PR with event + data tests - as I've confirmed it still reproduces (locally in Chrome 150, against main).

I've read the new contribution guidelines, including the section on unsolicited submissions, which is why I'm confirming the approach here before submitting any code. I also want to ackowledge that some of my earlier PRs were closed under that new policy, which I fully understand, and this is me trying to contribute the right way. 

Would you be open to a codepen link, and/or a video of the before/after on my lcoal machine before proceeding?
